packages/legend-studio/src/stores/__tests__/roundtrip/RelationalRoundtripTestData.ts

```diff
@@ -8978,3 +8978,266 @@ export const testRelationalAssociationMapping = [
+        },
+      ],
+    },
+    classifierPath: 'meta::pure::metamodel::type::Class',
+  },
+  {
+    path: 'test::Person',
+    content: {
+      _type: 'class',
+      name: 'Person',
+      package: 'test',
+      properties: [
+        {
+          multiplicity: {
+            lowerBound: 1,
+            upperBound: 1,
+          },
+          name: 'name',
+          type: 'String',
+        },
+      ],
+    },
+    classifierPath: 'meta::pure::metamodel::type::Class',
+  },
+  {
+    path: 'test::Firm_Person',
+    content: {
+      _type: 'association',
+      name: 'Firm_Person',
+      package: 'test',
+      properties: [
+        {
+          multiplicity: {
+            lowerBound: 1,
+            upperBound: 1,
+          },
+          name: 'employer',
+          type: 'test::Firm',
+        },
+        {
+          multiplicity: {
+            lowerBound: 0,
+          },
+          name: 'employees',
+          type: 'test::Person',
+        },
+      ],
+    },
+    classifierPath: 'meta::pure::metamodel::relationship::Association',
+  },
+  {
+    path: 'test::crossPropertyMappingWithLocalProperties',
+    content: {
+      _type: 'mapping',
+      associationMappings: [
+        {
+          _type: 'xStore',
+          association: 'test::Firm_Person',
+          propertyMappings: [
+            {
+              _type: 'xStorePropertyMapping',
+              crossExpression: {
+                _type: 'lambda',
+                body: [
+                  {
+                    _type: 'func',
+                    function: 'equal',
+                    parameters: [
+                      {
+                        _type: 'property',
+                        parameters: [
+                          {
+                            _type: 'var',
+                            name: 'this',
+                          },
+                        ],
+                        property: 'firmId',
+                      },
+                      {
+                        _type: 'property',
+                        parameters: [
+                          {
+                            _type: 'var',
+                            name: 'that',
+                          },
+                        ],
+                        property: 'id',
+                      },
+                    ],
+                  },
+                ],
+                parameters: [],
+              },
+              property: {
+                class: 'test::Person',
+                property: 'employer',
+              },
+              source: 'p',
+              target: 'f',
+            },
+          ],
+          stores: [],
+        },
+      ],
+      classMappings: [
+        {
+          _type: 'pureInstance',
+          class: 'test::Person',
+          id: 'p',
+          propertyMappings: [
+            {
+              _type: 'purePropertyMapping',
+              explodeProperty: false,
+              localMappingProperty: {
```

Comment:
I see we have a test here for local mapping property, but since it's a separate thing, can we separate this out to a new test instead of gobbling it up with another test?

Reply:
`Class test::Firm
{
  id: Integer[1];
  legalName: String[1];
}

Class test::Person
{
  name: String[1];
}

Association test::Firm_Person
{
  employer: test::Firm[1];
  employees: test::Person[*];
}


###Mapping
Mapping test::crossPropertyMappingWithLocalProperties
(
  test::Person[p]: Pure
  {
    ~src test::Person
    +firmId: Integer[1]: 1,
    name: $src.name
  }
  test::Firm[f]: Pure
  {
    ~src test::Firm
    id: $src.id,
    legalName: $src.legalName
  }

  test::Firm_Person: XStore
  {
    employer[p, f]: $this.firmId == $that.id
  }
)
`
Will split into 2

LocalMapping:
`Class test::Firm
{
  id: Integer[1];
  legalName: String[1];
}

Class test::Person
{
  name: String[1];
}

Association test::Firm_Person
{
  employer: test::Firm[1];
  employees: test::Person[*];
}


###Mapping
Mapping test::crossPropertyMappingWithLocalProperties
(
  test::Person[p]: Pure
  {
    ~src test::Person
    +firmId: Integer[1]: 1,
    name: $src.name
  }
)
`
XStore 
`Class test::Firm
{
  id: Integer[1];
  legalName: String[1];
}

Class test::Person
{
  name: String[1];
}

Association test::Firm_Person
{
  employer: test::Firm[1];
  employees: test::Person[*];
}


###Mapping
Mapping test::crossPropertyMappingWithLocalProperties
(
  test::Person[p]: Pure
  {
    ~src test::Person
    name: $src.name
  }
  test::Firm[f]: Pure
  {
    ~src test::Firm
    id: $src.id,
    legalName: $src.legalName
  }

  test::Firm_Person: XStore
  {
    employer[p, f]: $this.name == $that.legalName
  }
)
`

Reply:
split into two roundtrip tests